File: katas/6 kyu/break-camelcase/solution.py
# break-camelcase
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def solution(s):
  letters = 'abcdefghijklmnopqrstuvwxyz'.upper()
  _str = ''
  for x in s:
    if x in letters:
      _str += " "
    _str += x
  return _str
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def solution(s):
  return ''.join(' ' + c if c.isupper() else c for c in s)

import re
def solution(s):
  return re.sub('([A-Z])', r' \1', s)
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(solution("helloWorld"), "hello World")
test.assert_equals(solution("camelCase"), "camel Case")
test.assert_equals(solution("breakCamelCase"), "break Camel Case")

File: katas/6 kyu/break-camelcase/solution.js
// break-camelcase
/* ------------------------------- MY SOLUTION ------------------------------ */
function solution(string) {
  return string.split("").map(x => x === x.toUpperCase() ? ` ${x}` : x ).join("")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function solution(string) {
  return(string.replace(/([A-Z])/g, ' $1'));
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;
const chai = require("chai");
const assert = chai.assert;

describe("Sample Tests", () => {
  it("Should pass sample tests", () => {
    assert.strictEqual(solution('camelCasing'), 'camel Casing', 'Unexpected result')
    assert.strictEqual(solution('camelCasingTest'), 'camel Casing Test', 'Unexpected result')
  });
});